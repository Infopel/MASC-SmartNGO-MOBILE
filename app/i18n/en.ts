const translation = {
  common: {
    ok: 'OK!',
    exit: 'Exit',
    cancel: 'Cancel',
    back: 'Back',
    'select-an-item': 'Select an item',
    'date-placeholder': 'eg: 25/12/2021',
    save: 'Save',
    empty_list_message: 'Empty List',
    province: 'Province',
    district: 'District/Municipality',
    zone: 'Village/Neighborhood',
    'camera-add': 'Add',
    remove: 'remove',
    'camera-preview': 'Preview',
    thumbnail: 'Thumbnail',
    'preview-image': 'Preview',
    locality: 'Locality',
    project: 'Project',
    name: 'Name',
  },
  'context-menu': {
    'view-action-plan': 'View action plan',
    'view-monitoring-visit': 'View montoring visits',
  },
  login: {
    username: 'Username',
    password: 'Password',
    error: {
      timeout: 'Could not connect, check your internet connection',
      'cannot-connect': 'Could not connect, check your internet connection',
      canceled: 'Could not connect, check your internet connection',
      server: 'Could not connect, check your internet connection',
      unauthorized: 'You are not authenticated, please login',
      forbidden: 'Your connection has expired, please login again',
      'not-found': 'Your credentials are invalid, please try again',
    },
  },
  errors: {
    invalidEmail: 'Invalid email address.',
  },
  settings: {
    title: 'Settings',
  },
  'civic-incubator-details-participants-tab': {
    title: 'Participants',
    search: 'Search participant',
    'add-participant': 'Add participant',
    'participant-item': '%{item} participant',
  },
  'civic-incubator-details-events-tab': {
    title: 'Lectures',
    search: 'Search lecture',
    'add-participant': 'Add participant',
    'add-event': 'Create event',
  },
  'peace-ambassadors-and-vdo': {
    vdo: {
      title: 'VDO',
    },
    'peace-ambassadors': {
      title: 'Peace Ambassadors',
    },
    new: 'New',
  },
  'peace-ambassadors-and-vdo-details': {
    members: 'Members',
    actions: 'Actions',
    'add-member': 'Add member',
    'add-action': 'Add action',
    vdo: {
      title: 'VDO Details',
    },
    'peace-ambassadors': {
      title: 'Peace Ambassadors Details',
    },
  },
  'new-peace-ambassadors-and-vdo': {
    vdo: {
      title: 'New VDO',
    },
    'peace-ambassadors': {
      title: 'New Peace Ambassadors',
    },
    'group-code': 'Group Code',
    'date-of-constitution': 'Date of constitution',
    'number-of-active-members': 'Number of active members',
    'number-of-members': 'Number of Members',
  },
  home: {
    title: 'Home',
    vdo: 'VDO',
    'civic-incubators': 'Civic Incubators',
    'group-savings': 'Saving Groups',
    'peace-ambassadors': 'Peace Ambassadors',
    'my-tasks': 'My Tasks',
  },
  sync: {
    title: 'Synchronize',
  },
  stats: {
    title: 'Stats',
  },
  search: {
    title: 'Search',
  },
  'savings-group': {
    title: 'Savings Groups',
    'z_cycle-duration': 'Cycle duration: ',
    'cycle-duration': {
      zero: 'No cycle duration',
      one: 'Cycle duration: 1 day',
      others: 'Cycle duration: %{count} days',
    },
    'number-of-cycles': {
      zero: 'No cycles',
      one: '1 Cycle',
      others: '%{count} Cycles',
    },
    'z_number-of-cycles': ' Cycles',
  },
  'civic-incubators': {
    title: 'Civic Incubators',
    search: 'Search Civic Incubators',
    'item-label': 'Civic incubator item',
    'delete-item-label': 'Delete civic incubator item',
  },
  'new-civic-incubator': {
    title: 'New Civic incubator',
    productName: 'Civic incubator name',
    province: 'Province',
    district: 'District',
    locality: 'Locality',
    longitude: 'Longitude',
    latitude: 'latitude',
    'admin-post': 'Administrative Post',
    'date-constitution': 'Date of Constituition',
  },
  'civic-incubator-details': {
    title: 'Incubator Details',
  },
  'savings-group-details-monitoring-tab': {
    title: 'Monitoring',
    'add-monitoring': 'Add Monitoring',
  },
  'savings-group-details-members-tab': {
    title: 'Members',
    'add-member': 'Add Member',
    monitoring: 'monitor',
    quitting: 'quit',
  },
  'action-plan': {
    title: 'Action plan',
    add: 'Add action plan',
  },
  'monitoring-visits': {
    title: 'Monitoring visits',
    add: 'Add monitoring visit',
  },
  form: {
    error_retry:
      'Something went wrong, check if everything is ok and try again',
    'next-page': 'Next',
    'prev-page': 'Previous',
    submit: 'Finish',
    'savings-group-member-quitting': {
      title: 'Quitting',
      motivation: 'Motivation',
      date: 'Date',
    },
    cycles: {
      title: 'Cycles',
      'cycles-name': 'Name',
      'cycles-start-date': 'start Date',
      'cycles-end-date': 'End date',
    },
    'savings-group-member-monitoring': {
      title: 'Monitoring',
      'social-fund-contribuition': 'Has contribuited to social fund',
      'interest-value': 'Interest value',
      'loan-value': 'Loan value',
      'saved-value': 'Saved value',
      date: 'Date',
    },

    'monitoring-visits': {
      title: 'Monitoring visits',
      date: 'Date of visit',
      type: 'Type of visit',
      motive: 'Motive',
    },
    'report-action-plan-progress': {
      title: 'Report',
      'start-date': 'Real start date',
      'end-date': 'Real end date',
      'done-ratio': 'Progress percentage',
      observations: 'Observations',
    },
    'action-plan': {
      title: 'Action plan',
      'start-date': 'Start date',
      'end-date': 'End date',
      name: 'Name',
    },
    'savings-group': {
      title: 'New Savings Group',
      province: 'Province',
      district: 'District/Municipality',
      'select-province': 'Select a province',
      zone: 'Village/Neighborhood',
      'date-of-constitution': 'Date of constitution',
      'social-fund-value': 'Social fund value',
      'fine-value': 'Fine value',
      'interests-fee': 'Interest fee',
      name: 'Name',
      'number-of-active-members': 'Number of active members',
      'number-of-members': 'Number of Members',
      'number-of-cycles': 'Number of Cycles',
    },
    'civic-incubator-events': {
      title: 'New Lecture',
      realization_date: 'Realization Date',
      type: 'Section Type',
      type_other: 'Section Type',
      name: 'Section Name',
      subject: 'Subject',
      coordinator: 'Coordinator | Speaker',
      number_of_participants_male: 'Number of young participants (Male)',
      number_of_participants_female: 'Number of young participants (Female)',
      developed_actions: 'Developed actions',
      action_results: 'Action results',
      action_results_other: 'Action results',
      observations: 'Observations',
    },
    'civic-incubator-participant': {
      title: 'New Participant',
      'birth-date': 'Date of Birth',
      'first-name': 'First Name',
      'last-name': 'Last Name',
      gender: 'Gender',
      'education-level': 'Education level',
      'marital-status': 'Marital status',
      occupation: 'Occupation',
      conhecimentoIncubadora: 'Where did you heard about the civic incubator?',
      conhecimentoEspaco: 'Do you engage on the spaces?',
      participacaoEspaco: 'Engagement spaces',
      actividadePolitica: 'Do you envelve on politics?',
      observations: 'Comments',
    },
    'task-diary': {
      title: 'Task diary',
      'step-1': 'Task Details',
      name: 'Task name',
      'task-objetive': 'Task objetive',
      'date-start': 'Starting date',
      'date-end': 'Ending date',
      'spended-hours': 'Spended hours',
      'execution-percentage': 'Execution percentage',
      'step-2': 'Achived results and learned lessons',
      'meeting-description': 'Meeting Description',
      'imidiact-result': 'What was the imidiact result',
      'who-to-inform': 'Who would you like to inform/influence',
      'challenges-n-lesson': 'Faced challenges and learned lessons',
      'next-steps': 'Next steps',
      'masc-contribuition': 'Is there somo contibuition expected from MASC',
      'step-3': 'Means of evindence verification',
      'evidence-type': 'Type of evidence',
      'verification-methods': 'Verification methods',
      'number-of-males': 'Number of participants - males',
      'number-of-females': 'Number of participants - females',
      'generic-comments': 'Generic comments',
      'text-evidences': 'Evidences',
      'image-evidences': 'Evidences',
      'step-4': 'Indicators and Goals',
      indicator: 'Indicator',
      goal: 'Goal',
      'type-of-goal': 'Type of goal',
      'verification-source': 'Verification sources',
      realized: 'Realized',
      Evidences: 'Evidences',
      Upload: 'Upload',
    },
    'pa-or-vdo-member': {
      title: 'Peace Ambassadors and VDO',
      'first-name': 'First Name',
      'last-name': 'Last Name',
      occupation: 'Occupation',
      'education-level': 'Education level',
      'member-code': 'Member code',
      'marital-status': 'Marital status',
      function: 'Function',
      'active-member': 'Is Active Member',
      'why-gave-up': 'Why Gave Up',
    },
    'pa-or-vdo-action': {
      title: 'Register action',
      'realization-date': 'Realization date',
      type: 'Action type',
      name: 'Action name',
      goal: 'Action goal',
      coordinator: 'Action coordinator',
      observations: 'Observations',
      number_of_participants_male: 'Number of young participants (Male)',
      number_of_participants_female: 'Number of young participants (Female)',
    },

    'savings-group-monitoring': {
      title: 'Savings Group monitoring',
      cycle: 'Cycle',
      'gained-interest-savings': 'Gained  interests (Savings)',
      'gained-interest-social-fund': 'Gained interests (Social fund)',
      'cooperative-evolution': 'Evelved to cooperative?',
      'cooperative-type': 'type of cooperative',
      'main-problems': 'Main problems pointed out by the group',
    },
    'savings-group-member': {
      title: 'Savings Group member',
      'step-1': 'Beneficiary',
      'first-name': 'First Name',
      'last-name': 'Last Name',
      'birth-date': 'Date of Birth',
      'marital-status': 'Marital status',
      'education-level': 'Education level',
      'number-of-children': 'Number of children',
      'living-with-parents': 'Living with parents?',
      zone: 'Village/Neighborhood',
      'available-wealth': 'What are your goods (wealth) ?',
      'active-member': 'Is Active Member',
      'why-gave-up': 'Why Gave Up?',
      'step-2': 'Savings',
      cycle: 'Cycle',
      'savings-value': 'Savings value',
      'loan-value': 'Loan value',
      'loan-interests': 'loan interest rate',
      'fine-value': 'Fine value',
      'social-fund-value': 'Social fund value',
      'business-activity': 'Income business activity',
      'savings-application': 'Savings application',
      'previous-credit-benefit':
        'Have ever benefited from other form of credit?',
    },
  },
  'my-tasks': {
    title: 'My tasks',
    pending: 'Pending',
    finished: 'Finished',
    'pending-task': 'Pending task',
    'pending-task-item': 'Pending task item',
    'finished-task': 'Finished task',
    'finished-task-item': 'Finished task item',
    'search-tasks': 'Search tasks',
  },
  error: {
    title: 'Something went wrong!',
    friendlySubtitle:
      "This is the screen that your users will see in production when an error is thrown. You'll want to customize this message (located in `app/i18n/en.json`) and probably the layout as well (`app/screens/error`). If you want to remove this entirely, check `app/app.tsx` for the <ErrorBoundary> component.",
    traceTitle: 'Error from %{name} stack',
    reset: 'RESET APP',
  },
  storybook: {
    placeholder: 'Placeholder',
    field: 'Field',
  },
};

export type TranslationLang = typeof translation;
export default translation;
