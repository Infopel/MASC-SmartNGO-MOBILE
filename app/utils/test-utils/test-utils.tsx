import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

