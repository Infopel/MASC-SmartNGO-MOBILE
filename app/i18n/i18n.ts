import * as Localization from "expo-localization"
//@ts-ignore
import {I18n} from "i18n-js/dist/require"
import en from "./en"
import pt from "./pt"
export const i18n =  new I18n({en,pt})
i18n.enableFallback = true
i18n.defaultLocale = "en"
// i18n.missingBehavior = "error";

i18n.locale = Localization.locale || "en"

/**
 * Builds up valid keypaths for translations.
 * Update to your default locale of choice if not English.
 */
type DefaultLocale = typeof en
export type TxKeyPath = RecursiveKeyOf<DefaultLocale>

type RecursiveKeyOf<TObj extends Record<string, any>> = {
  [TKey in keyof TObj & string]: TObj[TKey] extends Record<string, any>
  //@ts-ignore  
  ? `${TKey}` | `${TKey}.${RecursiveKeyOf<TObj[TKey]>}`
    : `${TKey}`
}[keyof TObj & string]
