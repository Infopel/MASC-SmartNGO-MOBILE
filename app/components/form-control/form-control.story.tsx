import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { FormControl } from "./form-control.component"

storiesOf("FormControl", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <FormControl style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
