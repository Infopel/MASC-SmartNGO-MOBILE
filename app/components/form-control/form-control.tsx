import { translate, TxKeyPath } from "i18n"
import { useField } from "formik"
import { isNil } from "lodash"
import { FormControl as NBFormControl, IBoxProps } from "native-base"
import * as React from "react"


export interface FormControlProps extends IBoxProps {
  children: React.ReactNode
  fieldName: string
  label: TxKeyPath
  helperText?: TxKeyPath
}

/**
 * Describe your component here
 */
export const FormControl = function FormControl(props: FormControlProps) {
  const { children, label, helperText, fieldName, ...rest } = props
  const [field, { error }] = useField(fieldName)

  return (
    <NBFormControl isInvalid={!isNil(error)} {...rest}>
      <NBFormControl.Label>{translate(label)}</NBFormControl.Label>
      {children}
      {helperText && <NBFormControl.HelperText>{translate(helperText)}</NBFormControl.HelperText>}
      <NBFormControl.ErrorMessage>{error}</NBFormControl.ErrorMessage>
    </NBFormControl>
  )
}
