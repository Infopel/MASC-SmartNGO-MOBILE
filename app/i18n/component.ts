import {TranslateOptions} from 'i18n-js/typings';
import logger from 'utils/logger';
import {TxKeyPath} from './i18n';
import {translate} from './translate';

export interface ITranslatable {
  /**
   * Optional options to pass to i18n. Useful for interpolation
   * as well as explicitly setting locale or translation fallbacks.
   */
  txOptions?: TranslateOptions;
}
export interface ITranslatableText extends ITranslatable {
  /**
   * Text which is looked up via i18n.
   */
  tx?: TxKeyPath;

  /**
   * Optional options to pass to i18n. Useful for interpolation
   * as well as explicitly setting locale or translation fallbacks.
   */
  txOptions?: TranslateOptions;

  /**
   * The text to display if not using `tx` or nested components.
   */

  children?: React.ReactNode;
}

export function translateComponent({
  tx,
  children,
  txOptions,
}: ITranslatableText): React.ReactNode | undefined {
  const i18nText = tx && translate(tx, txOptions);

  if (typeof i18nText !== 'string' && i18nText !== undefined) {
    console.warn('Invalid translated text', {i18nText});
    return children;
  }
  return children || i18nText;
}
