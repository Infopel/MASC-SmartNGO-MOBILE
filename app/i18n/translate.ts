import {TranslateOptions} from 'i18n-js/typings';
import {i18n} from './i18n';
import {TxKeyPath} from './i18n';

/**
 * Translates text.
 *
 * @param key The i18n key.
 */
export function translate(key: TxKeyPath, options?: TranslateOptions) {
  if (!key) throw new Error('Key cannot be nil or empty');
  const trans = i18n.t(key, options);

  if (typeof trans !== 'string')
  throw new Error("The inserted translation key should not map to and object");
  
  return trans;
}
