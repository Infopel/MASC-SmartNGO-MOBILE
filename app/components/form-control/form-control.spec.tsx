import { Formik } from "formik"
import { TxKeyPath } from "i18n/i18n"
import { translate } from "i18n/translate"
import * as React from "react"
import { View } from "react-native"
import { render } from "utils/test-utils/test-utils"
import { FormControl } from "./form-control"


describe("<FormControl>",()=>{
  it("should render", ()=>{
    const submit = jest.fn()
    const label:TxKeyPath = 'civic-incubators.delete-item-label'
    const {getByTestId, getByText, toJSON} = render(<Formik onSubmit={submit} initialValues={{ control: undefined }}><FormControl {...{ testID:"tatabas",fieldName: 'control', label }}>    <View />
    </FormControl>
    </Formik>)
    expect(getByTestId("tatabas")).toBeTruthy()
    expect(getByText(translate(label))).toBeTruthy()
    expect(toJSON()).toMatchSnapshot()
    
  })
  it("should show helperText", ()=>{
    const submit = jest.fn()
    const helperText:TxKeyPath = 'civic-incubator-details.title'
    const label:TxKeyPath = 'civic-incubators.delete-item-label'
    const {getByText, toJSON} = render(<Formik onSubmit={submit} initialValues={{ control: undefined }}><FormControl {...{ testID:"tatabas",fieldName: 'control', helperText, label }}>    <View />
    </FormControl>
    </Formik>)
    
    expect(getByText(translate(helperText))).toBeTruthy()
    expect(toJSON()).toMatchSnapshot()
    
  })
  
  it("should show error", ()=> {
    const submit = jest.fn()
    const label = 'civic-incubators.delete-item-label'
    const errorMessage = "this is a error message"
    const {getByText, toJSON} = render(<Formik onSubmit={submit} initialErrors={{
      control:errorMessage
    }} initialValues={{ control: undefined }}><FormControl {...{ testID:"tatabas",fieldName: 'control', label }}>    <View />
    </FormControl>
    </Formik>)

    expect(getByText(errorMessage)).toBeTruthy()
    expect(toJSON()).toMatchSnapshot()

    
    
  })
})


