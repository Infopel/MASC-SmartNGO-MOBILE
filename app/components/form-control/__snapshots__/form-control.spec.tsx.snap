// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormControl> should render 1`] = `
<View
  dataSet={Object {}}
  isInvalid={false}
  style={
    Object {
      "width": "100%",
    }
  }
  testID="tatabas"
>
  <View
    dataSet={Object {}}
    feedbackId="field-2-feedback"
    hasFeedbackText={true}
    hasHelpText={false}
    helpTextId="field-2-helptext"
    isDisabled={false}
    isInvalid={false}
    isReadOnly={false}
    isRequired={false}
    labelId="field-2-label"
    nativeID="field-2-label"
    setHasFeedbackText={[Function]}
    setHasHelpText={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 4,
        "marginTop": 4,
      }
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#737373",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Delete civic incubator item
    </Text>
  </View>
  <Text
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
  >
        
  </Text>
  <View />
</View>
`;

exports[`<FormControl> should show error 1`] = `
<View
  dataSet={Object {}}
  isInvalid={true}
  style={
    Object {
      "width": "100%",
    }
  }
  testID="tatabas"
>
  <View
    dataSet={Object {}}
    feedbackId="field-8-feedback"
    hasFeedbackText={true}
    hasHelpText={false}
    helpTextId="field-8-helptext"
    isDisabled={false}
    isInvalid={true}
    isReadOnly={false}
    isRequired={false}
    labelId="field-8-label"
    nativeID="field-8-label"
    setHasFeedbackText={[Function]}
    setHasHelpText={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 4,
        "marginTop": 4,
      }
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#737373",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Delete civic incubator item
    </Text>
  </View>
  <Text
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
  >
        
  </Text>
  <View />
  <View
    dataSet={Object {}}
    feedbackId="field-8-feedback"
    hasFeedbackText={true}
    hasHelpText={false}
    helpTextId="field-8-helptext"
    isDisabled={false}
    isInvalid={true}
    isReadOnly={false}
    isRequired={false}
    labelId="field-8-label"
    nativeID="field-8"
    setHasFeedbackText={[Function]}
    setHasHelpText={[Function]}
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    <View
      dataSet={Object {}}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        dataSet={Object {}}
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#dc2626",
            "fontFamily": undefined,
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 18,
            "textDecorationLine": undefined,
          }
        }
      >
        this is a error message
      </Text>
    </View>
  </View>
</View>
`;

exports[`<FormControl> should show helperText 1`] = `
<View
  dataSet={Object {}}
  isInvalid={false}
  style={
    Object {
      "width": "100%",
    }
  }
  testID="tatabas"
>
  <View
    dataSet={Object {}}
    feedbackId="field-5-feedback"
    hasFeedbackText={true}
    hasHelpText={true}
    helpTextId="field-5-helptext"
    isDisabled={false}
    isInvalid={false}
    isReadOnly={false}
    isRequired={false}
    labelId="field-5-label"
    nativeID="field-5-label"
    setHasFeedbackText={[Function]}
    setHasHelpText={[Function]}
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginBottom": 4,
        "marginTop": 4,
      }
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#737373",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Delete civic incubator item
    </Text>
  </View>
  <Text
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": undefined,
        "color": "#171717",
        "fontFamily": undefined,
        "fontSize": 14,
        "fontStyle": "normal",
        "fontWeight": "400",
        "letterSpacing": 0,
        "lineHeight": 21,
        "textDecorationLine": undefined,
      }
    }
  >
        
  </Text>
  <View />
  <View
    dataSet={Object {}}
    feedbackId="field-5-feedback"
    hasFeedbackText={true}
    hasHelpText={true}
    helpTextId="field-5-helptext"
    isDisabled={false}
    isInvalid={false}
    isReadOnly={false}
    isRequired={false}
    labelId="field-5-label"
    nativeID="field-5-feedback"
    setHasFeedbackText={[Function]}
    setHasHelpText={[Function]}
    style={
      Object {
        "marginTop": 8,
      }
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#737373",
          "fontFamily": undefined,
          "fontSize": 12,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 18,
          "textDecorationLine": undefined,
        }
      }
    >
      Incubator Details
    </Text>
  </View>
</View>
`;
